include/libsurvive/survive_api.h

```diff
@@ -0,0 +1,63 @@
+#include "survive_types.h"
```

Comment:
I think that this API might best be handled as a layer above the core libsurvive.  I.e. a separate make target that consumes libsurvive.  That way, you can implement lots of features that make libsurvive easy to use without cluttering or adding complexity to the core libsurvive.

Reply:
In practice I would want to see how much more clutter the simple api stuff actually adds -- if its just one C file, it seems tolerable. 

The downside to having two separate libraries would be some confusion on end users about which one they actually need and where they actually need it; especially if it was structured so one SO had a dependency on the other. Also, if the user wanted to switch from one API to the other, or possibly use both (presumably in different modes or something; not concurrently), it'd present some complications .